charts/argocd-applicationset/templates/deployment.yaml

```
@@ -59,6 +59,13 @@ spec:
             - name: webhook
               containerPort: 7000
               protocol: TCP
+          {{- if .Values.env }}
+          env:
+          {{- toYaml .Values.env | nindent 12 }}
+          {{- end }}
+          {{- with .Values.envFrom }}
+          envFrom: {{- toYaml . | nindent 12 }}
+          {{- end }}
```

Let's call it `extraEnv` and use the `with` keyword
```suggestion
          {{- with .Values.extraEnv }}
          env:
            {{- toYaml . | nindent 12 }}
          {{- end }}
          {{- with .Values.envFrom }}
          envFrom:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

---

updated both variable to avoid confusion. I am not sure what is the standard, but argo-cd uses `env` & `envFrom`, but all others only have `extraEnv`.

---

Yup, unfortunately both variants exists in this helm repo.
Maybe we should refactor some charts to follow a standard.